internal/cli/writer.go

```diff
@@ -157,5 +159,26 @@ func (fw *fileWriter) write(filename string, p []byte) (int, error) {
 	if fw.writer != nil {
```

Comment:
I suggest converting the whole function as follow:

```go
func (fw *fileWriter) write(filename string, p []byte) (int, error) {
	// if run in check mode return exit 1
	if fw.check {
		f, err := os.ReadFile(filename)
		if err != nil {
			return 0, err
		}

		// check for changes and print changed file
		if !bytes.Equal(f, p) {
			return 0, fmt.Errorf("%s is out of date", filename)
		}

		fmt.Printf("%s is up to date.\n", filename)
		return 0, nil
	}

	if fw.writer != nil {
		return fw.writer.Write(p)
	}

	fmt.Printf("%s updated successfully.\n", filename)
	return len(p), os.WriteFile(filename, p, 0644)
}
```

Reply:
Consecutively this test case can also be added in `writer_test.go` at the end of `TestFileWriter`:

```go
		"ModeReplaceOutOfDate": {
			file:     "mode-replace.md",
			mode:     "replace",
			check:    true,
			template: OutputTemplate,
			begin:    outputBeginComment,
			end:      outputEndComment,
			writer:   nil,

			expected: "",
			wantErr:  true,
			errMsg:   "testdata/writer/mode-replace.md is out of date",
		},
```

Note that you also need to add `check bool` to the struct and `check: false,` to all the other test cases.

Reply:
@rickhlx did you want to finish off your changes here? or would you rather me adding these new changes instead? Either way is great!

Reply:
Sorry! Been running around a bit. I'll take care of these changes, no problem. Thanks for the comments!